src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/AgreementsApiServiceImpl.scala

```diff
@@ -266,21 +266,20 @@ final case class AgreementsApiServiceImpl(
   ): Seq[AttributeRegistry.Attribute] =
     filterIds.flatMap(id => registryAttributes.attributes.find(_.id == id))
 
-  def enhanceTenantAttribute(
-    registryAttributes: MgmtAttributesResponse,
-    tenantAttribute: TenantManagement.TenantAttribute
-  ): TenantAttribute = {
-    val certified = tenantAttribute.certified.map(a => (a, registryAttributes.attributes.find(_.id == a.id))).collect {
-      case (a1, Some(a2)) => a1.toApi(a2.name)
-    }
-    val declared  = tenantAttribute.declared.map(a => (a, registryAttributes.attributes.find(_.id == a.id))).collect {
-      case (a1, Some(a2)) => a1.toApi(a2.name)
-    }
-    val verified  = tenantAttribute.verified.map(a => (a, registryAttributes.attributes.find(_.id == a.id))).collect {
-      case (a1, Some(a2)) => a1.toApi(a2.name)
-    }
+  def enhanceTenantAttributes(
+    tenantAttributes: Seq[TenantManagement.TenantAttribute],
+    registryAttributes: MgmtAttributesResponse
+  ): Seq[TenantAttribute] = {
 
-    TenantAttribute(declared = declared, certified = certified, verified = verified)
-  }
+    val registryAttributesMap = registryAttributes.attributes.map(a => (a.id, a)).toMap
```

Comment:
```suggestion
    val registryAttributesMap = registryAttributes.attributes.fproductLeft(_.id).toMap
```

See `fproductLeft` [here](https://www.javadoc.io/doc/org.typelevel/cats-docs_2.13/latest/cats/Functor.html)